includes/database.inc

```diff
@@ -79,12 +377,20 @@ function update_sql($sql) {
  *   The properly-prefixed string.
  */
 function db_prefix_tables($sql) {
-  global $db_prefix;
+  global $db_prefix, $db_mysql8_reserved_keywords;
+
+  //need to escape reserved keywords for MySQL 8
+  if(db_is_version('mysql', '8.0.0')) {
+    $sql = array_reduce($db_mysql8_reserved_keywords, function ($original_sql, $keyword) {
+      return str_replace('{' . $keyword . '}', '`{' . $keyword . '}`', $original_sql);
+    }, $sql);
```

Comment:
This appears to be wrapping the `{table}` before the prefix is added, which doesn't seem quite right - I think it would make the most sense to check if the table with the prefix added is a reserved keyword.

It's relatively unlikely, but let's say someone used the db prefix 'read_' and then had a table called 'write': that would lead to an actual table name of 'read_write' which is a reserved keyword, but isn't found by the check.

Or, what if it just wrapped all table names in backticks?

Reply:
Sorry for delay with answer.

> This appears to be wrapping the {table} before the prefix is added, which doesn't seem quite right - I think it would make the most sense to check if the table with the prefix added is a reserved keyword.

Thank you for pointing to this!

I've changed db_prefix_tables() function for it to warp table names with backticks after adding prefixes.

> Or, what if it just wrapped all table names in backticks?

Actually I do not like this for such "deep core" code. This change can break some other code relying on "unwrapped" table names in queries. Let we change such core logic only when it really unavoidable (as for reserved keywords only in this case).